### PR TITLE
Issue #11712: Support yield in FallThrough

### PR DESCRIPTION
**Pull request by KENNYSOFT**
Issue #11712: Support yield in FallThrough

Diff Regression config: https://gist.githubusercontent.com/KENNYSOFT/9ba5f1a7d14011feca79047b333cac59/raw/7c36ed41edb2c19652e3a6e1ec14ec93968680fc/my_check-FallThrough.xml

### CODE REVIEW

**Comment by KENNYSOFT** [comment]
GitHub, generate report

**Comment by romani** [comment]
GitHub, generate report

**Comment by romani** [comment]
I didn't found problems in diff report.

Ok to merge

**Review by romani — APPROVED** [review]

File: src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages_fr.properties [hunk 1]
```diff
@@ -16,8 +16,8 @@ equals.noEquals=La définition de la méthode ''hashCode()'' doit toujours être
 equals.noHashCode=La définition de la méthode ''equals()'' doit toujours être accompagnée de la définition de la méthode ''hashCode()''.
 equalsIgnoreCase.avoid.null=Les chaînes littérales devraient être sur le côté gauche d''une comparaison ''equalsIgnoreCase''.
 explicit.init=L''initialisation explicite de la variable ''{0}'' à la valeur ''{1}'' est inutile, c''est la valeur par défaut pour ce type.
-fall.through=Le cas précédent du "switch" ne contient pas de break, return, throw ou continue.
-fall.through.last=Le dernier cas du "switch" ne contient pas de break, return, throw ou continue.
+fall.through=Le cas précédent du "switch" ne contient pas de break, return, yield, throw ou continue.
+fall.through.last=Le dernier cas du "switch" ne contient pas de break, return, yield, throw ou continue.
```
Comment (strkkk) [inline src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages_fr.properties:20]: why changes are only in French localization? Other languages dont need it?

**Review by strkkk — CHANGES_REQUESTED** [review]

File: src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages_fr.properties (hunk 1 above)
Comment (KENNYSOFT) [reply src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages_fr.properties:20]: Just found out Deutsch needs also, will amend. Except for two languages, translation is based on "Fall through from previous branch of the switch statement" so don't need it. Examples follow:

* [es] Caída desde la etiqueta anterior en la sentencia switch.
* [pt] Continuação (fallthrough) vindo da ramificação anterior da instrução switch.
* [tr] ''switch'' ifadesinin bir önceki durumundan aşağı düşme mevcut (''break'' kullanılmamış olabilir).

Comment (KENNYSOFT) [reply src/main/resources/com/puppycrawl/tools/checkstyle/checks/coding/messages_fr.properties:20]: Resolved in b2c18a8c08cddd27455d3520f4cb2d060c1a0041

**Review by strkkk — APPROVED** [review]

**Pull request merged**